traffic_ops/app/db/migrations/SQUASH.md

```diff
@@ -0,0 +1,46 @@
+# Squashing database migrations
+
+Each major release of Apache Traffic Control combines database migrations from previous ATC releases into [`create_tables.sql`](https://github.com/apache/trafficcontrol/blob/master/traffic_ops/app/db/create_tables.sql).
+
+For example, suppose the latest version of Apache Traffic Control is 147.5.8 and contains these migrations:
+* `1_my-migration.up.sql`
+* `1_my-migration.down.sql`
+* `3_another-migration.up.sql`
+* `3_another-migration.down.sql`
+
+And suppose the ATC [`master`](https://github.com/apache/trafficcontrol/commits/master) branch contains these migrations:
+* `1_my-migration.up.sql`
+* `1_my-migration.down.sql`
+* `3_another-migration.up.sql`
+* `3_another-migration.down.sql`
+* `4_migration-name.up.sql`
+* `4_migration-name.down.sql`
+* `9_add-column-to-table.up.sql`
+* `9_add-column-to-table.down.sql`
+
+1. In order to prepare database migrations for the next major release, in this case, ATC 148.0.0, migrations `1` and `3` should be collapsed into `create_tables.sql` and migrations `4` and `9` should remain in [`traffic_ops/app/db/migrations/`](https://github.com/apache/trafficcontrol/tree/master/traffic_ops/app/db/migrations/).
+
+2. Note that `3` is the migration version version of the last up/down migration set. Find the definition for `LastSquashedMigrationVersion` in [`traffic_ops/app/db/admin.go`](https://github.com/apache/trafficcontrol/blob/master/traffic_ops/app/db/admin.go) and change it to `3`.
```

Comment:
version repeated twice.

Reply:
Is there a way to automate this manual step? For eg: I haven't read create_tables.sql,  but when create_tables.sql runs and squashes/collapses previous migration, last migration date is stored. To remember to always update this date, I feel a user may easily forget.

Reply:
> version repeated twice.

Removed one *version* in 84f2871c99

Reply:
> Is there a way to automate this manual step? For eg: I haven't read create_tables.sql, but when create_tables.sql runs and squashes/collapses previous migration, last migration date is stored. To remember to always update this date, I feel a user may easily forget.

Added `squash_migrations.sh`, which automates squashing migrations from the last ATC release, in 3de5dfdb08. We will lose `create_tables.sql` idempotency, which is really only an issue when two Traffic Ops instances are running the Postinstall script simultaneously, which is what #4984 was about. IMO this is not a big issue, but I filed #6106 so we don't lose track of it.